eureka-client/src/main/java/com/netflix/appinfo/providers/MyDataCenterInstanceConfigProvider.java

```diff
@@ -6,6 +6,9 @@
 import com.netflix.discovery.DiscoveryManager;
 import com.netflix.discovery.EurekaNamespace;
 
+import javax.inject.Singleton;
+
+@Singleton
```

Comment:
@Singleton is unnecessary here.   Making the Provider @Singleton does not make MyDataCenterInstanceConfig a singleton.  You need to do that when you specify the binding.  This should also be a provider for EurekaInstanceConfig and not MyDataCenterInstanceConfig.

bind(EurekaInstanceConfig.class).toProvider(MyDataCenterInstanceConfigProvider.class).in(Scopes.SINGLETON)